### ISSUE TITLE
nixd: url prefetching

### ISSUE THREAD

**Issue by inclyc** [nix-community/nixd#461]
Provide URL prefetchers. Automatically fill hash section.

```
fetchurl {

}
```

**Comment by Origami404** [comment]
Should this feature be implemented as a "code action" that requires manual triggering? The files to be downloaded may be quite large.

**Comment by inclyc** [comment]
> Should this feature be implemented as a "code action" that requires manual triggering? The files to be downloaded may be quite large.

Yes. I'd like to do this in Code Lens, with `workspace/executeCommand`

**Comment by inclyc** [comment]
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeLens

**Comment by inclyc** [comment]
@MangoIV  https://github.com/nix-community/nixd/issues/536#issue-2375239352

> offer to augment the passed attribute set with the hash computed from prefetching it.

Note that things like `fetchFromGithub` are actually FODs, not just "fetchers", and require some evaluation. 
I'm currently on design-review about how to balance between evaluation and fetching by idioms.

I also searched nixpkgs and see there are quite some expressions like:

```
fetchFromGitHub {
  rev = "v${version}";
}
``` 

And thus "version" needs to be evaluated. And...

`finalAttrs.version` may needs some cross-file analysis.
